### PR TITLE
add schema.prisma

### PR DESCRIPTION
**Pull request by theborgh**
- Used strings as ids even on postgres for consistency with the existing User schema in t3 boilerplate
- t3 boilerplate also has email as optional. I left it optional so it doesn't accidentally break nextAuth

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -6,33 +6,82 @@ generator client {
...
 }
 
-model Example {
+model Project {
+    id         String   @id @default(cuid())
+    name       String
+    owner      User     @relation("owner", fields: [ownerId], references: [id])
+    ownerId    String
+    developers User[]   @relation("developers")
+    bugs       Bug[]
```
Comment (1qk1) [inline prisma/schema.prisma:23]: `bug` is not needed here.

If we add `project` in the `Bug` (like you have correctly done), we can get all objects that are in the selected project (One to Many relationship).

In One to Many relationships (One project, many bugs), we set the foreign key on the _many_ model (In this case `Bug`).

This also applies to developers.

Comment (1qk1) [inline prisma/schema.prisma:23]: Is the `bugs` property needed here?

In SQL databases, if we add `project` in the `Bug` (like you have correctly done), we can get all objects that are in the selected project (One to Many relationship).

In One to Many relationships (One project, many bugs), we set the foreign key on the _many_ model (In this case `Bug`).

**Review by 1qk1 — COMMENTED** [review]
Please double check if we need the relations in both models, since in SQL databases, we only need it in 1 of the 2.

Edit: Never mind, it's probably a shortcut of prisma, to get all posts

**Comment by 1qk1** [comment]
Also don't mind the failing checks, I haven't finished setting up the project on netlify.

**Review by 1qk1 — APPROVED** [review]

**Comment by theborgh** [comment]
> Please double check if we need the relations in both models, since in SQL databases, we only need it in 1 of the 2.
> Edit: Never mind, it's probably a shortcut of prisma, to get all related items

Thanks for the feedback! You're right, from my understanding this is prisma syntax that does not appear in the actual table (only the prisma entries with primitives like Int, String etc are reflected in the db)

**Review by mehdi-zibout — APPROVED** [review]
Thanks Dario

**Comment by 1qk1** [comment]
@mehdi-zibout Just a heads up, the owner of the PR has to merge after approval.

**Pull request merged**